### PR TITLE
Issue 5038 - [2d mvp] Milestone 4 & 5

### LINKED ISSUE #5038: [2D MVP] Milestone 4
### Description
Base issue for milestone 4
dependencies:

- [x] https://github.com/3drepo/3drepo.io/issues/5084
- [x] https://github.com/3drepo/3drepo.io/issues/5039
- [x] https://github.com/3drepo/3drepo.io/issues/5040
- [x] https://github.com/3drepo/3drepo.io/issues/5043
- [x] https://github.com/3drepo/3drepo.io/issues/5103

### Product issue
https://github.com/3drepo/3D-Repo-Product-Team/issues/380

### Related Resources
[Navigation
](https://www.figma.com/design/uTWnaYZ8IyoTn2i2xKtdBo/2D---3D?node-id=11-9027&t=Z3TmeDiRW11yggwo-0)

[Show 2d Pins](https://www.figma.com/design/uTWnaYZ8IyoTn2i2xKtdBo/2D---3D?node-id=11-11332&t=Z3TmeDiRW11yggwo-0)

### PR DESCRIPTION
**Pull request by sanmont3drepo**
This fixes #5038 

#### Description
- This pr shows the camera  and pins positions on a calibrated drawing
- The user can drop pins in 2d and 3d

### CODE REVIEW

**Review by carmenfan — CHANGES_REQUESTED** [review]
@sanmont3drepo  as discussed i'll pause and wait till the fix on the 2D viewer before further testing

**Review by carmenfan — CHANGES_REQUESTED** [review]

File: frontend/assets/icons/viewer/camera.svg.tsx
Comment (carmenfan) [inline frontend/assets/icons/viewer/camera.svg.tsx:1]: Sorry this is more to do with milestone 1, but can we tweak this menu slightly so it's more inline with containers:

image

- Move `Upload` to the top and rename it to `Upload new revision`
- Rename `Calibrate` to `Calibrate latest revision`

Comment (carmenfan) [inline frontend/assets/icons/viewer/camera.svg.tsx:1]: The icon we're using on 2D viewer is incorrect for intersection/node snapping:
image

Should be the same one (and colour) as the corner snapping icon on 3D:
image

[the remark this answers is not in the document]
Comment (sanmont3drepo) [reply frontend/assets/icons/viewer/camera.svg.tsx:1]: @sebjf  can you check what the problem might be with this?

[the remark this answers is not in the document]
Comment (sebjf) [reply frontend/assets/icons/viewer/camera.svg.tsx:1]: Hi @sanmont3drepo, @carmenfan, I am unable to reproduce this (see below!). I have tried on both my high DPI and low DPI monitor. @carmenfan is it repeatable for you? 

https://github.com/user-attachments/assets/34ae7f3c-7e9c-48c0-8bf3-a8a1ffe201fe

I have tried placing pins, different sizes, zoom levels & Firefox. 

Looking at it the only thing that jumps to mind is the inertia emulation going over some limit in one dimension. If that is turned off, does the issue go away?

[the remark this answers is not in the document]
Comment (sanmont3drepo) [reply frontend/assets/icons/viewer/camera.svg.tsx:1]: Can you assume that that case can happen and make it work for that case?

[the remark this answers is not in the document]
Comment (sanmont3drepo) [reply frontend/assets/icons/viewer/camera.svg.tsx:1]: @sebjf another piece of  info: I can only reproduce it if after I switch drawings.

[the remark this answers is not in the document]
Comment (carmenfan) [reply frontend/assets/icons/viewer/camera.svg.tsx:1]: @sebjf Yes it's repeatable and it seems to only happen if I've changed drawing

[the remark this answers is not in the document]
Comment (sebjf) [reply frontend/assets/icons/viewer/camera.svg.tsx:1]: OK I can reproduce now, looking into it..

Comment (carmenfan) [inline frontend/assets/icons/viewer/camera.svg.tsx:1]: Hovering on the out of camera icon  should change the cursor just like the in view camera gizmo

**Review by carmenfan — DISMISSED** [review]

**Review by carmenfan — APPROVED** [review]

**Pull request merged**